python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED_test.py

```diff
@@ -0,0 +1,363 @@
+            "resourceName": user_name,
+            "ARN": "arn:aws:iam::0123456789012:user/{}".format(user_name),
+            "awsRegion": "global",
+            "availabilityZone": "Not Applicable",
+            "configurationStateMd5Hash": "",
+            "resourceCreationTime": "2018-08-30T13:25:10.000Z"
+        },
+        "notificationCreationTime": "2018-09-21T09:24:28.962Z",
+        "messageType": "ConfigurationItemChangeNotification",
+        "recordVersion": "1.3"
+    }
+
+
+def build_role_configuration_item(attached_policies_arns, role_name="test-role", path="/"):
+    return {
+        "configurationItemDiff": None,
+        "configurationItem": {
+            "relatedEvents": [],
+            "relationships": [
+                {
+                    "resourceId": "ABCDEFGHI12JKL4MNO5PQ",
+                    "resourceName": arn.rsplit("/")[-1],
+                    "resourceType": "AWS::IAM::Policy",
+                    "name": "Is attached to CustomerManagedPolicy"
+                } for arn in attached_policies_arns
+            ],
+            "configuration": {
+                "path": path,
+                "roleName": role_name,
+                "roleId": "ABCDEFGHI12JKL4MNO5PQ",
+                "arn": "arn:aws:iam::0123456789012:role{Path}{Name}".format(Path=path, Name=role_name),
+                "createDate": "2018-08-30T13:25:10.000Z",
+                "assumeRolePolicyDocument": "{}",
+                "instanceProfileList": [],
+                "rolePolicyList": [],
+                "attachedManagedPolicies": [
+                    {
+                        "policyName": arn.rsplit("/")[-1],
+                        "policyArn": arn
+                    } for arn in attached_policies_arns
+                ],
+                "permissionsBoundary": None
+            },
+            "supplementaryConfiguration": {},
+            "tags": {},
+            "configurationItemVersion": "1.3",
+            "configurationItemCaptureTime": "2018-09-07T11:24:32.476Z",
+            "configurationStateId": 12345678901,
+            "awsAccountId": "0123456789012",
+            "configurationItemStatus": "OK",
+            "resourceType": "AWS::IAM::Role",
+            "resourceId": "ABCDEFGHI12JKL4MNO5PQ",
+            "resourceName": role_name,
+            "ARN": "arn:aws:iam::0123456789012:role/{}".format(role_name),
+            "awsRegion": "global",
+            "availabilityZone": "Not Applicable",
+            "configurationStateMd5Hash": "",
+            "resourceCreationTime": "2018-08-30T13:25:10.000Z"
+        },
+        "notificationCreationTime": "2018-09-21T09:24:28.962Z",
+        "messageType": "ConfigurationItemChangeNotification",
+        "recordVersion": "1.3"
+    }
+
+
+rule = __import__('IAM_POLICY_REQUIRED')
+
+class TestPolicyRequired(unittest.TestCase):
+
+    rule_parameters = '{"policyArns":["arn:aws:iam::aws:policy/AdministratorAccess"], "exceptionList": ""}'
```

Comment:
the RDK boilerplate uses `rule_parameters = json.loads(event['ruleParameters'])` to load the rule parameters therefore this comes in as a list. Similarly, if you pass a list type in through CFN (e.g. a parameter of type comma separated list, it will be a Python List when passed to the rule without any string splitting required etc.

I could change it to a string, but it would mean parsing would need to be added in the Lambda function custom code rather than handled by the RDK boilerplate. what do you think?

Reply:
```event['ruleParameters']``` comes as a dir. The value of each key of this dir must be a string. Currently your value are list, so it should not be working if you try to deploy it in the Config service.

Incorrect:
```'{"policyArns":["arn:aws:iam::aws:policy/AdministratorAccess"], "exceptionList": ""}'```

Correct:
```'{"policyArns":"[arn:aws:iam::aws:policy/AdministratorAccess]", "exceptionList": ""}'```

Reply:
@jongogogo updated!